### PR TITLE
Clean up SqliteItemRepository

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 1]
```diff
@@ -3380,93 +3380,61 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement statem
...
-                    programAttribtues.Add("(IsMovie is null OR IsMovie=@IsMovie)");
+                    whereClauses.Add("IsMovie=@IsMovie");
                 }
 
-                if (statement != null)
-                {
-                    statement.TryBind("@IsMovie", true);
-                }
-
-                whereClauses.Add("(" + string.Join(" OR ", programAttribtues) + ")");
```
Comment (mark-monteiro) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:3452]: Why is this removed?

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 2]
```diff
@@ -3315,7 +3315,7 @@ private bool IsAlphaNumeric(string str)
 
             for (int i = 0; i < str.Length; i++)
             {
-                if (!(char.IsLetter(str[i])) && (!(char.IsNumber(str[i]))))
+                if (!char.IsLetter(str[i]) && (!char.IsNumber(str[i])))
```
Comment (JustAMan) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:3318]:
```suggestion
                if (!char.IsLetter(str[i]) && !char.IsNumber(str[i]))
```

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 3]
```diff
@@ -3380,93 +3380,61 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement statem
...
-                    statement.TryBind("@IsLocked", query.IsLocked);
-                }
+                statement?.TryBind("@IsLocked", query.IsLocked);
             }
 
             var tags = query.Tags.ToList();
             var excludeTags = query.ExcludeTags.ToList();
 
-            if (query.IsMovie ?? false)
+            if (query.IsMovie == true)
```
Comment (JustAMan) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:3413]: why not just
```suggestion
            if (query.IsMovie)
```

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs [hunk 4]
```diff
@@ -3686,98 +3609,74 @@ private List<string> GetWhereClauses(InternalItemsQuery query, IStatement statem
...
+                    clause.Append("TrailerTypes like ")
+                        .Append(paramName)
+                        .Append(" OR ");
+                    statement?.TryBind(paramName, "%" + trailerTypes[i] + "%");
                 }
-                var clause = "(" + string.Join(" OR ", clauses) + ")";
-                whereClauses.Add(clause);
+
+                // Remove last " OR "
+                clause.Length -= 4;
```
Comment (JustAMan) [inline Emby.Server.Implementations/Data/SqliteItemRepository.cs:3660]:
```suggestion
                clause.Length -= " OR ".Length;
```
or something similar

File: MediaBrowser.Controller/Persistence/IItemRepository.cs [hunk 5]
```diff
@@ -25,7 +25,7 @@ public interface IItemRepository : IRepository
         /// </summary>
         /// <param name="id">The identifier.</param>
         /// <param name="cancellationToken">The cancellation token.</param>
```
Comment (JustAMan) [inline MediaBrowser.Controller/Persistence/IItemRepository.cs:27]:
```suggestion
```

File: Emby.Server.Implementations/Data/SqliteItemRepository.cs (hunk 3 above)
Comment (Bond-009) [reply Emby.Server.Implementations/Data/SqliteItemRepository.cs:3413]: It's a `bool?`, that doesn't work
